host/libhackrf/src/hackrf.h

```diff
@@ -165,274 +385,816 @@ typedef struct hackrf_device hackrf_device;
+ */
 extern ADDAPI int ADDCALL hackrf_set_freq(hackrf_device* device, const uint64_t freq_hz);
 
+/**
+ * Set the center frequency via explicit tuning
+ * 
+ * Center frequency is set to \f$f_{center} = f_{ID} + k\cdot f_{LO}\f$ where \f$k\in\left\{-1; 0; 1\right\}\f$, depending on the value of @p path. See the documentation of @ref rf_path_filter for details
+ * 
+ * @param device device to tune
+ * @param if_freq_hz tuning frequency of the MAX2837 transceiver IC in Hz. Must be in the range of 2150-2750MHz
+ * @param lo_freq_hz tuning frequency of the RFFC5072 mixer/synthesizer IC in Hz. Must be in the range 84.375-5400MHz, defaults to 1000MHz. No effect if @p path is set to @ref RF_PATH_FILTER_BYPASS
+ * @param path filter path for mixer. See the documentation for @ref rf_path_filter for details
+ * @return @ref HACKRF_SUCCESS on success or @ref hackrf_error variant  
+ */
 extern ADDAPI int ADDCALL hackrf_set_freq_explicit(
 	hackrf_device* device,
 	const uint64_t if_freq_hz,
 	const uint64_t lo_freq_hz,
 	const enum rf_path_filter path);
+	
 
-/* 2-20Mhz - either as a fraction, i.e. freq 20000000hz divider 2 -> 10Mhz or as plain old 10000000hz (double) */
+// TODO: is this info correct?
+/**
+ * Set sample rate explicitly
+ * 
+ * Sample rate should be in the range 2-20MHz, with the default being 10MHz. Lower & higher values are technicly possibly, but the performance is not guaranteed.
+ * 
+ * This function sets the sample rate by specifying a clock frequency in Hz and a divider, so the resulting sample rate will be @p freq_hz / @p divider.
+ * This function also sets the baseband filter bandwidth to a value <= 0.75 * samplerate, so a call to @ref hackrf_set_baseband_filter_bandwidth should be made after it.
+ * 
+ * @param device device to configure
+ * @param freq_hz sample rate base frequency in Hz
+ * @param divider frequency divider. Must be in the range 1-31
+ * @return @ref HACKRF_SUCCESS on success or @ref hackrf_error variant  
+ */
 extern ADDAPI int ADDCALL hackrf_set_sample_rate_manual(
 	hackrf_device* device,
 	const uint32_t freq_hz,
 	const uint32_t divider);
 
+/**
+ * Set sample rate
+ * 
+ * Sample rate should be in the range 2-20MHz, with the default being 10MHz. Lower & higher values are technicly possibly, but the performance is not guaranteed.
+ * This function also sets the baseband filter bandwidth to a value <= 0.75 * samplerate, so a call to @ref hackrf_set_baseband_filter_bandwidth should be made after it.
+ * 
+ * @param device device to configure
+ * @param freq_hz sample rate frequency in Hz. Should be in the range 2-20MHz
+ * @return @ref HACKRF_SUCCESS on success or @ref hackrf_error variant  
+ */
 extern ADDAPI int ADDCALL hackrf_set_sample_rate(
 	hackrf_device* device,
 	const double freq_hz);
 
-/* external amp, bool on/off */
+/**
+ * Enable/disable 14dB RF amplifier
+ * 
+ * Enable / disable the ~14dB RF RX/TX amplifiers U13/U25 via controlling switches U9 and U14.
```

Comment:
I was a bit confused there, but the [FAQ section on the docs](https://hackrf.readthedocs.io/en/latest/faq.html#what-gain-controls-are-provided-by-hackrf) says 14dB. Shall I open a separate issue on that or fix it in this PR?

Reply:
Yeah, unfortunately that error has crept into a lot of places. I wrote a bit more about it in https://github.com/greatscottgadgets/hackrf/issues/1059#issuecomment-1060038293.

I think it would be best to keep this PR focused on the Doxygen side, and correct other documentation in another PR.

Reply:
Opened issue #1247